src/examples/ImageScaleUp.cpp

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All rights reserved.
+
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+
+#include "Common.h"
+
+/************************************************************************/
+/* Drawing Commands                                                     */
+/************************************************************************/
+
+static tvg::Picture* pPicture = nullptr;
+
+void tvgDrawCmds(tvg::Canvas* canvas)
+{
+    if (!canvas) return;
+
+    //Original
+    auto picture = tvg::Picture::gen();
+    pPicture = picture.get();
+
+    if (picture->load(EXAMPLE_DIR"/scaleup.png") == tvg::Result::Success) {
+        if (canvas->push(move(picture)) != tvg::Result::Success) return;
+    }
+}
+
+void tvgUpdateCmds(tvg::Canvas* canvas, float progress)
+{
+    if (!canvas) return;
+
+    auto scale = 1.0f;
+
+    if (progress > 0.875f) scale = 4.0f;
+    else if (progress > 0.75f) scale = 3.0f;
+    else if (progress > 0.625f) scale = 2.5f;
+    else if (progress > 0.5f) scale = 2.0f;
+    else if (progress > 0.375f) scale = 1.75f;
+    else if (progress > 0.25f) scale = 1.5f;
+    else if (progress > 0.125f) scale = 1.25f;
+    else scale = 1.0f;
```

Comment:
Instead of l.49-58 I'd write:
```
    auto scale = 1.0f;
    auto step = 0.125f;

    if (progress > 0.875f) scale = 4.0f;
    else if (progress > 0.75f) scale = 3.0f;
    else scale = 1.0f + 2 * step * (int)(progress / step);
```

Reply:
@mgrudzinska thanks for suggestion, but both are not too complex actually.
Intuitive code is sometimes better to keep in examples for readers.